### PR TITLE
Prep for 4.3.8 release

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
Updates changelog and version references for 4.3.8.

Previous version: #7749
Release Code: TODO
New Changelog: TODO

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Review by paulbalandan — APPROVED** [review]

File: user_guide_src/source/changelogs/v4.3.8.rst [hunk 1]
```diff
@@ -1,26 +1,14 @@
 Version 4.3.8
 #############
 
-Release Date: Unreleased
+Release Date: August 25th, 2023
```
Comment (paulbalandan) [inline user_guide_src/source/changelogs/v4.3.8.rst:4]: Do we need this as ordinal?

Comment (kenjis) [reply user_guide_src/source/changelogs/v4.3.8.rst:4]: Oh, thanks! No.

Comment (kenjis) [reply user_guide_src/source/changelogs/v4.3.8.rst]: Removed.

**Review by paulbalandan — APPROVED** [review]

**Comment by kenjis** [comment]
Okay, all checks have passed!

**Pull request merged**